### PR TITLE
Rename catalog field in applyApp/publishCatalog pipeline step

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/17277

### CODE REVIEW

**Comment by loganhz** [comment]
LGTM

File: apis/project.cattle.io/v3/pipeline_types.go [hunk 1]
```diff
@@ -272,17 +272,17 @@ type ApplyYamlConfig struct {
...
 type PublishCatalogConfig struct {
-	Path      string `json:"path,omitempty" yaml:"path,omitempty"`
-	Catalog   string `json:"catalog,omitempty" yaml:"catalog,omitempty"`
-	Version   string `json:"version,omitempty" yaml:"version,omitempty"`
-	GitURL    string `json:"gitUrl,omitempty" yaml:"gitUrl,omitempty"`
-	GitBranch string `json:"gitBranch,omitempty" yaml:"gitBranch,omitempty"`
-	GitAuthor string `json:"gitAuthor,omitempty" yaml:"gitAuthor,omitempty"`
-	GitEmail  string `json:"gitEmail,omitempty" yaml:"gitEmail,omitempty"`
+	Path            string `json:"path,omitempty" yaml:"path,omitempty"`
+	CatalogTemplate string `json:"catalogTemplate,omitempty" yaml:"catalogTemplate,omitempty"`
```
Comment (alena1108) [inline apis/project.cattle.io/v3/pipeline_types.go]: Does it mean that we are breaking api compatibility?

Comment (loganhz) [reply apis/project.cattle.io/v3/pipeline_types.go]: This is a new feature in 2.2

Comment (alena1108) [reply apis/project.cattle.io/v3/pipeline_types.go]: Ok, all good then

**Comment by alena1108** [comment]
LGTM

**Pull request merged**